oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -1648,142 +1648,145 @@ int mca_spml_ucx_put_all_nb(void *dest, const void *source, size_t size, long *c
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_INT();
 }
 
 /* This routine is not implemented */
 void mca_spml_ucx_wait_until_all(void *ivars, int cmp, void
         *cmp_value, size_t nelems, const int *status, int datatype)
 {
-    return ;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_wait_until_any(void *ivars, int cmp, void
         *cmp_value, size_t nelems, const int *status, int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_wait_until_some(void *ivars, int cmp, void
         *cmp_value, size_t nelems, size_t *indices, const int *status, int
         datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */ 
 void mca_spml_ucx_wait_until_all_vector(void *ivars, int cmp, void
         *cmp_values, size_t nelems, const int *status, int datatype)
 {
-    return ;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_wait_until_any_vector(void *ivars, int cmp, void
         *cmp_value, size_t nelems, const int *status, int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_wait_until_some_vector(void *ivars, int cmp, void
         *cmp_value, size_t nelems, size_t *indices, const int *status, int
         datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */
 int mca_spml_ucx_test_all(void *ivars, int cmp, void *cmp_value,
         size_t nelems, const int *status, int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_INT();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_test_any(void *ivars, int cmp, void *cmp_value,
         size_t nelems, const int *status, int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */
 size_t mca_spml_ucx_test_some(void *ivars, int cmp, void *cmp_value,
         size_t nelems, size_t *indices, const int *status, int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    RUNTIME_SHMEM_NON_IMPLEMENTED_API_ABORT_RET_SIZE_T();
 }
 
 /* This routine is not implemented */
-int mca_spml_ucx_test_all_vector(void *ivars, int cmp, void
-        *cmp_values, size_t nelems, const int *status, int datatype)
+int mca_spml_ucx_test_all_vector(void *ivars, int cmp, void *cmp_values,
+                                 size_t nelems, const int *status,
+                                 int datatype)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
```

Comment:
@roiedanino can you please explain how this line leads to a warning? I understand the waring for size_t but not for int.

Reply:
This one does not, but it's incorrect to return a return code as it doesn't match specifications to do so, the caller won't expect a return code and won't interpret it as such

Reply:
We could internally return `ssize_t` to be able to encode errors. That would resolve the warning.

Reply:
So we will need to add `int* return_value` to all of those functions to be able to return both return code and the value that need to be returned to the caller by specifications

Reply:
@devreal @MamziB Wdyt?

Reply:
I thought these functions return `size_t` in the standard? If you replace `size_t` with `ssize_t` (`signed size_t`) you can return a negative error code and the index/size required by the standard. That will silence the compiler warnings I pointed out (which complained about an unsigned value being tested for negative values)